docs/components/zeebe/technical-concepts/batch-operations.md

```diff
@@ -6,117 +6,424 @@ description: How distributed batch operations work in Zeebe engine.
 
 Zeebe supports executing certain operations as batch operations:
 
-- Resolve incidents
-- Migrate process instances
-- Modify process instances
-- Cancel process instances
+- **Cancel process instances** (`CANCEL_PROCESS_INSTANCE`)
+- **Migrate process instances** (`MIGRATE_PROCESS_INSTANCE`)
+- **Modify process instances** (`MODIFY_PROCESS_INSTANCE`)
+- **Resolve incidents** (`RESOLVE_INCIDENT`)
 
-## How do they work?
+## Overview
+
+Batch operations allow you to perform actions on multiple process instances efficiently. This is particularly useful when you need to:
+
+- Cancel hundreds of process instances after discovering a critical bug
+- Migrate thousands of running instances to a new process version
+- Modify process instances to skip or add activities across multiple instances
+- Resolve incidents across multiple process instances simultaneously
+
+Instead of manually operating on each instance, batch operations distribute the work across your cluster, processing instances in parallel while maintaining system stability.
+
+### How it works at a high level
+
+When you create a batch operation, you provide filter criteria (such as process definition, version, or state) rather than a specific list of instances. The system then:
+
+1. Distributes the operation across all partitions in your cluster
+2. Each partition queries the secondary database to find matching instances
+3. Processes instances in parallel across partitions
+4. Tracks progress and aggregates results
+
+This distributed approach enables efficient processing of large numbers of instances while maintaining cluster performance.
```

Comment:
🔧 You could additionally add that a batch operation does not block a partition, but that the partition concurrently executes the batch operation on process instances with all other processing that occurs on the partition. In the same way as that two process instances are executed concurrently on the same partition.